### DISCUSSION TITLE
Use multiple dbcontext in the same query

### DISCUSSION THREAD

**Discussion by Bandude** [ChilliCream/hotchocolate#4283] [Q&A] [answered]
In order to maintain code, we've decided to separate the dbcontext into multiple files per schema. 

We're using the Poold DbContext Factory in the start-up class. 
![image](https://user-images.githubusercontent.com/48070856/135171183-91ac16c6-61d0-405c-88d9-246058fa4c7c.png)

We have a scenario where we would like to pull data from two different schemas and combine those into a custom object on the return.  

The problem is we can't seem to add two different scoped services. 
We could potentially just move everything into a single dbcontext, but we're concerned about maintainability long term. 

this is not a working example of course, but might give you an idea of what we're trying to accomplish. 
![image](https://user-images.githubusercontent.com/48070856/135172151-f6a9dd60-0de0-4b6a-96e6-45f4add63ecf.png)

I'm new to pooled dbcontext so I'm sure I'm missing something. 

Any help would be greatly appreciated.

**Comment by tobias-tengler** [comment, marked as answer]
From your example you are missing the `UseDbContext` attribute for the second DbContext. What's happening is that the UseDbContext middleware creates an instance of the DbContext using the factory and pushes it onto the local state in order to make it available to the resolver. The ScopedServiceAttribute is then used to retrieve the instance from the local state. When using the ScopedServiceAttribute you alwqys need to have a call to the UseDbContext middleware with the same DbContext type.

**Reply by Bandude** [reply]
Thanks, should have caught that.   I appreciate your help!
